Gardenlet in local gardener fails to create seed due to IP CIDR conflicts

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
-->
/area dev-productivity
/kind bug

**What happened**:
Following this guide https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md, it failed on step `make gardener-up` as no seed was created. In fact, gardenlet is crashing with this error message
```
panic: failed during bootstrapping: incorrect node network specified in seed configuration (cluster node="192.168.16.2" vs. config="172.18.0.0/16")
```

**What you expected to happen**:
To be able to create local gardener environment

**How to reproduce it (as minimally and precisely as possible)**:
Follow the guide from https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md

**Anything else we need to know?**:
It is still working, if I checkout `v1.57.0` tag.

**Environment**:

- Gardener version: afa6db45466a5f1c171a3848ef0f86168464d92c
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

--- a/docs/deployment/getting_started_locally.md
+++ b/docs/deployment/getting_started_locally.md
@@ -18,6 +18,9 @@ Based on [Skaffold](https://skaffold.dev/), the container images for all require
   > If you plan on following the optional steps to [create a second seed cluster](#optional-setting-up-a-second-seed-cluster), the required resources will be more - at least `10` CPUs and `18Gi` memory.
   Additionally, please configure at least `120Gi` of disk size for the Docker daemon.
   > Tip: With `docker system df` and `docker system prune -a` you can cleanup unused data.
+- Make sure the `kind` docker network is using the CIDR `172.18.0.0/16`.
+  - If the network does not exist, it can be created with `docker network create kind --subnet 172.18.0.0/16`
+  - If the network already exists, the CIDR can be checked with `docker network inspect kind  | jq '.[].IPAM.Config[].Subnet'`. If it is not `172.18.0.0/16`, delete the network with `docker network rm kind` and create it with the command above.
 
 ## Setting up the KinD cluster (garden and seed)
 

--- a/docs/development/getting_started_locally.md
+++ b/docs/development/getting_started_locally.md
@@ -19,6 +19,9 @@ The Gardener components, however, will be run as regular processes on your machi
 
   Additionally, please configure at least `120Gi` of disk size for the Docker daemon.
   > Tip: With `docker system df` and `docker system prune -a` you can cleanup unused data.
+- Make sure the `kind` docker network is using the CIDR `172.18.0.0/16`.
+  - If the network does not exist, it can be created with `docker network create kind --subnet 172.18.0.0/16`
+  - If the network already exists, the CIDR can be checked with `docker network inspect kind  | jq '.[].IPAM.Config[].Subnet'`. If it is not `172.18.0.0/16`, delete the network with `docker network rm kind` and create it with the command above.
 - Make sure that you increase the maximum number of open files on your host:
   - On Mac, run `sudo launchctl limit maxfiles 65536 200000`
   - On Linux, extend the `/etc/security/limits.conf` file with

